src/main/java/core/basesyntax/figures/Figure.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.figures;
+
+import core.basesyntax.ColorSupplier;
+
+public abstract class Figure implements FigureEvaluation, IsDrawable {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private String color;
+
+    {
+        this.color = colorSupplier.getRandomColor();
+    }
```

Comment:
ColorSuppler should be used by FigureSupplier to generate figure and set required color (random or default)